### ISSUE TITLE
[Feature Request] Allow to order plugin in each dalay type

### ISSUE THREAD

**Issue by l2eused** [alangrainger/obsidian-lazy-plugins#4]
I have one problem with **commander** and **unicode search** plugins in android phone. The problem is when I set my 2 plugins above in same delay (long delay for example), my **unicode search** command icon can't load in toolbar (I use commander to custom my unicode search command for mobile toolbar) but for **advanced table** (I also use as command in mobile toolbar too) it doesn't. So I guess it because order of loading is alphabetically order so when **commander** plugin is loaded, the unicode search command is not yet load and cause error command (question icon in mobile toolbar). So I want to request this feature to fix my problem here.

TLDR; order loading to fix some plugins that need to run before/after others plugin.

my idea:
- adding section to order plugin to load before/after normal order

for example
```
**plugin list:**
- AA
- BB
- CC
- DD
- FF
- EE

**set**
AA as long delay
BB as long
CC as Instant
DD as long
FF as short
EE as long

**normal loading order**
CC FF AA BB DD EE

**want to set**
DD, AA as last
EE as first

so in order list should be

> set this plugin to first in order (sort by order in this box)
> `EE`

> set this plugin to last in order (sort by order in this box)
> `DD`
> `AA`

**in this case, order is**
CC FF EE BB DD AA
```

as above example, just add new 2 insert box that is **order as first** and **order as last**

the priority would be:
1. delay set (instant, short, long)
2. first order box
3. normal order (alphabet)
4. last order

and run as
1. instant first
2. instant normal
3. instant last
4. short first
5. short normal
.
.
.

This isn't main setup for user, just use it only for few plugins that need to be order. (Should be in advanced settings)

**Comment by alangrainger** [comment]
This is a duplicate of #2 , so I will close this one and keep discussion over there.

The answer is in the very first comment there, but I will add here as well:

> @alangrainger adds:
> 
> If there's enough interest I will make a companion plugin to Lazy Loader. It's pretty simple - the companion plugin would have a settings page with a draggable list of plugins so you can re-order them however you like.
> 
> Then it would hook into the plugin enabled event and rearrange the icons to your specified order whenever any plugin loads.

**Comment by alangrainger** [comment]
I plan to add this in the next release or so, if I can find out a very lean way to add UI dragging/reordering.

**Comment by alangrainger** [comment]
Thanks for the suggestion :+1:  A bit awkward UX for the user, but at least it's a workable option if nothing else presents itself.

**Comment by hollowhemlock** [comment]
### Ideas 
1. I like "load after" over "depends on"    - more transparent, doesn't add dependency implication
	- some setups benefit from loading order without being dependent (See Other Thoughts below)
2. "depends on"/"loads after" needs to sync independent of the "separate desktop/mobile configuration" setting so it doesn't need to be configured twice.
3. I think decoupling enabled plugin state from delay settings is a good idea.
4. I think enabled plugin state and delay settings would benefit from independent "separate desktop/mobile configuration" toggles. 

### Other Thoughts
I worked on the tasks plugin to try and get multiple auto suggestions to work between plugins in unison. My setup is Natural Language Dates > Tasks > Various Complements. The greedier plugin is last.
